releasenotes/notes/new-deprecation-utilities-066aff05e221d7b1.yaml

```diff
@@ -0,0 +1,8 @@
+---
+features:
+  - |
+    Added the function ``qiskit.util.deprecation.add_deprecation_to_docstring()``.
+    It will rewrite the function's docstring to include a Sphinx ``.. deprecated::` directive
+    so that the deprecation shows up in docs and with ``help()``. The deprecation decorators
+    from ``qiskit.util.deprecation`` call ``add_deprecation_to_docstring()`` already for you;
+    but you can call it directly if you are using different mechanisms for deprecations.
```

Comment:
I guess this sentence is only true after #9676.  As long as we also merge that before 0.24, we should be ok, otherwise we'll need to remember to remove it.

Reply:
Agreed. I'm planning to put up a PR immediately after this that applies this helper to `deprecate_arguments()` and `deprecate_function()`. I reverted that from this PR to make it easier to review

And I have a couple PRs queued to finish this deprecation thing so I can finish by helping @woodsp-ibm with the algorithms pending deprecation change. Trying my best to land it all well before 0.24 :)